hangups/http_utils.py

```diff
@@ -5,57 +5,126 @@
 import collections
 import logging
 
-from hangups import exceptions
+from hangups import exceptions, channel
 
 logger = logging.getLogger(__name__)
 CONNECT_TIMEOUT = 30
 REQUEST_TIMEOUT = 30
 MAX_RETRIES = 3
 
-FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body',
-                                                         'cookies'])
+FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body'])
 
 
-@asyncio.coroutine
-def fetch(method, url, params=None, headers=None, cookies=None, data=None,
-          connector=None):
-    """Make an HTTP request.
+class ClientSession(aiohttp.ClientSession):
```

Comment:
``hangups.channel.Channel._longpoll_request()`` uses ``.request()`` of the new ``http_utils.ClientSession``, which need to be bridge then as well.

As [``aiohttp.ClientSession.get()``](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client.py#L478..L483
) calls the private ``_request`` method directly instead of calling the public one ``request``, the proxy and header must to be added in the private method.